packages/stats/src/collector.ts

```diff
@@ -0,0 +1,74 @@
+import * as Parser from '@tracespace/parser'
+import {Stats} from './types'
+
+/**
+ * Board stats collector.
+ *
+ * @category StatsCollector
+ */
+export interface StatsCollector {
+  /** Collect stats of a parsed AST */
+  collect(root: Parser.Root): Stats
+}
+
+/**
+ * {@linkcode StatsCollector} factory and the primary export of the library.
+ *
+ * @example
+ * ```ts
+ * import {createParser} from '@tracespace/parser'
+ * import {createStatsCollector} from '@tracespace/stats'
+ *
+ * const parser = createParser()
+ * const statsCollector = createStatsCollector()
+ *
+ * parser.feed(...)
+ *
+ * const tree = parser.results()
+ * const stats = statsCollector.collect(tree)
+ * ```
+ *
+ * @category StatsCollector
+ */
+export function createStatsCollector(): StatsCollector {
+  return {collect}
+
+  function collect(root: Parser.Root): Stats {
+    const stats: Stats = {
+      tools: new Map<string, Parser.ToolShape>(),
+      drillsPerTool: new Map<string, number>(),
+      totalDrills: 0,
```

Comment:
This exports a bunch of drill specific tools, but:

1. The function is called a "stats collector", not a "drill stats collector"
2. The function does not check `root` to see if it's `filetype: 'drill'`

Since this PR is targeting drill stats, I'd say addressing item (2) for now is probably sufficient